File: hermes-server/app/models/machine.py
from sqlalchemy import Column, String, Integer, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
import uuid
from app.database import Base

class Machine(Base):
    __tablename__ = "machines"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String, ForeignKey("users.id"), nullable=False)
    name = Column(String, nullable=True)
    address = Column(String, nullable=False)  # IP:Port
    created_at = Column(Integer, default=lambda: int(datetime.utcnow().timestamp()))

    # Relationships
    user = relationship("User", back_populates="machines")
    agents = relationship("Agent", back_populates="machine")

File: hermes-app/lib/data/models/machine.dart
class Machine {
  final String id;
  final String? name;
  final String address;
  final String? apiKey;
  final int createdAt;

  Machine({
    required this.id,
    this.name,
    required this.address,
    this.apiKey,
    required this.createdAt,
  });

  factory Machine.fromJson(Map<String, dynamic> json) {
    return Machine(
      id: json['id'] as String,
      name: json['name'] as String?,
      address: json['address'] as String,
      apiKey: json['api_key'] as String?,
      createdAt: json['created_at'] as int,
    );
  }
}
